recipes/gdal/post_3.5.0/conandata.yml

```diff
@@ -1,20 +1,41 @@
 sources:
-  "3.7.0":
-    url: "https://github.com/OSGeo/gdal/releases/download/v3.7.0/gdal-3.7.0.tar.gz"
-    sha256: "5a806d759f403a15bbbf8a14ecc6947071afc5ab91e5abaef0d11d1d2d16bf94"
-  "3.5.2":
+  "3.8.2":
+    url: "https://github.com/OSGeo/gdal/releases/download/v3.8.2/gdal-3.8.2.tar.gz"
+    sha256: "e914690d2694671d75bab15b451a42c2f115d7267a5b600eb361e94b58a089d6"
+  "3.7.3":
+    url: "https://github.com/OSGeo/gdal/releases/download/v3.7.3/gdal-3.7.3.tar.gz"
+    sha256: "f66161e10b8b89a8a541cd760cd36d490114ed3f020a26db1489a6154db5d2be"
+  "3.5.3":
     url: "https://github.com/OSGeo/gdal/releases/download/v3.5.2/gdal-3.5.2.tar.gz"
     sha256: "fbd696e1b2a858fbd2eb3718db16b14ed9ba82521d3578770d480c74fe1146d2"
-  "3.5.1":
-    url: "https://github.com/OSGeo/gdal/releases/download/v3.5.1/gdal-3.5.1.tar.gz"
-    sha256: "7c4406ca010dc8632703a0a326f39e9db25d9f1f6ebaaeca64a963e3fac123d1"
 patches:
-  "3.7.0":
-    - patch_file: "patches/3.7.0/0-replace-find-package.patch"
-      base_path: "source_subfolder"
-  "3.5.2":
-    - patch_file: "patches/3.5.2/0-replace-find-package.patch"
-      base_path: "source_subfolder"
-  "3.5.1":
-    - patch_file: "patches/3.5.1/0-replace-find-package.patch"
-      base_path: "source_subfolder"
+  "3.8.2":
+    - patch_file: "patches/3.8.1/0-replace-find-package.patch"
+      patch_description: "Use custom version of find_package() for Conan deps"
+      patch_type: "conan"
+    - patch_file: "patches/3.8.1/1-fix-recursive-targets.patch"
+      patch_description: "Fix CMake failure due to recursive CMakeDeps Conan targets"
+      patch_type: "conan"
+    - patch_file: "patches/3.7.3/3-iconv-compatibility.patch"
+      patch_description: "Fix iconv() signature incompatibility on some platforms"
+      patch_type: "portability"
```

Comment:
It's an unfixed bug upstream. I linked to a related issue in the patch notes. I intend to submit it upstream as well, but it might need to be fixed a bit better - with a compilation check in CMake and an ifdef.

Reply:
Actually, it was already being handled correctly in the upstream code. The compilation check was simply failing for some reason. I dropped the patches and forced the compilation check value to the universally compatible non-const value for now. I'll see if I can maybe fix the compilation checks instead.